src/data/buffer/basic.lean

```diff
@@ -42,18 +42,120 @@ lemma append_list_mk_buffer  :
 by ext x : 1; simp [array.to_buffer,to_list,to_list_append_list];
    induction xs; [refl,skip]; simp [to_array]; refl
 
+@[simp] lemma to_buffer_to_list (b : buffer α) : b.to_list.to_buffer = b :=
+begin
+  cases b,
+  simp [to_list,to_array,list.to_buffer],
+  congr, simp, refl, apply array.to_list_to_array
+end
+
+@[simp] lemma to_list_to_buffer (l : list α) : l.to_buffer.to_list = l :=
+begin
+  induction l with hd tl hl, refl,
+  simp [list.to_buffer,append_list],
```

Comment:
Please enclose the separate branches of `induction` in curly brackets

Reply:
I think it's also OK to just add braces around `refl`, showing that it's a separate goal that is quickly dispensed with. Then the second branch can be left outside quotes as the main goal. This can save some indentation.

Happened to notice this though:
```suggestion
  simp [list.to_buffer, append_list],
```

Reply:
These proofs were copied out wholesale from the proof body of the equivalence below. I will reformat them according to the style guide.